include/subsystems/odometry/odometry_nwheel.h

```diff
@@ -0,0 +1,255 @@
+#pragma once
+
+// These are required for Eigen to compile
+// https://www.vexforum.com/t/eigen-integration-issue/61474/5
+#undef __ARM_NEON__
+#undef __ARM_NEON
+#include <Eigen/Dense>
+
+#include "../core/include/subsystems/custom_encoder.h"
+#include "../core/include/subsystems/odometry/odometry_base.h"
+#include "../core/include/utils/math_util.h"
+
+/**
+ * OdometryNWheel
+ *
+ * This class handles the code for an N-pod odometry setup, where there are N <WHEELS> free spinning omni wheels
+ * (dead wheels) placed in any known configuration on the robot.
+ *
+ * Example of a possible wheel configuration:
+ *
+ *  +Y   ---------------
+ *  ^    |    ===      |
+ *  |    |             |
+ *  |    | ||   O      |
+ *  |    | ||          |
+ *  |    |      ===    |
+ *  |    ---------------
+ *  |
+ *  +-------------------> + X
+ *
+ * Where O is the center of rotation. The robot will monitor the changes in rotation of these wheels, use this to
+ * calculate a pose delta, then integrate the deltas over time to determine the robot's position.
+ *
+ * This is a "set and forget" class, meaning once the object is created, the robot will immediately begin
+ * tracking it's movement in the background.
+ *
+ * https://rit.enterprise.slack.com/files/U04112Y5RB6/F080M01KPA5/predictperpindiculars2.pdf
+ * 2024-2025 Notebook: Entries/Software Entries/Localization/N-Pod Odometry
+ *
+ * @author Jack Cammarata, Richie Sommers
+ * @date Nov 14 2024
+ *
+ */
+
+/**
+ * tracking_wheel_cfg_t holds all the specifications for a single tracking wheel
+ * The units for x, y, and radius will determine the units of the position estimate
+ */
+typedef struct {
+  double x;         /**< x position of the center of the wheel */
+  double y;         /**< y position of the center of the wheel */
+  double theta_rad; /**< angle between wheel direction and x axis in the robot frame */
+  double radius;    /**< radius of the wheel */
+} tracking_wheel_cfg_t;
+
+template <int WHEELS> class OdometryNWheel : public OdometryBase {
+public:
+  /**
+   * Construct a new Odometry N Wheel object
+   *
+   * @param encoders std::array containing each CustomEncoder
+   * @param wheel_configs std::array containing each tracking_wheel_cfg
+   * @param is_async true to constantly run in the background
+   */
+  OdometryNWheel(
+    const std::array<CustomEncoder, WHEELS> &encoders, const std::array<tracking_wheel_cfg_t, WHEELS> wheel_configs,
+    vex::inertial *imu, bool is_async
+  )
+      : OdometryBase(is_async), imu(imu), encoders(encoders) {
```

Comment:
do you initialize the old_wheel_angles variable anywhere? if not, probably initialzied to zero here

Reply:
same with rotation_offset

Reply:
changed these, at some point we should probably go through the codebase and replace all "pos" with "pose"

Reply:
fair. it still looks like angle_offset and old_wheel_angles are uninitialized though. unless im missing it. can you explicityl initialize them to 0